include/agl/aglGPUMemBlock.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <container/seadPtrArray.h>
+#include <heap/seadDisposer.h>
+#include <hostio/seadHostIONode.h>
+#include <math/seadMathCalcCommon.h>
+#include <prim/seadTypedBitFlag.h>
+#include <thread/seadCriticalSection.h>
+#include "aglGPUCommon.hpp"
+
+namespace agl {
+
+namespace detail {
+class MemoryPool;
+class MemoryPoolHeap;
+};  // namespace detail
+
+class GPUMemBlockBase {
+public:
+    GPUMemBlockBase(sead::Heap* p_heap);
```

Comment:
```suggestion
    explicit GPUMemBlockBase(sead::Heap* p_heap);
```
(we probably don't want implicit conversions for this one)